### PR TITLE
fix(MToken): handle allowance in burn

### PR DESCRIPTION
**Pull request by PierrickGT**

### CODE REVIEW

File: src/MToken.sol [hunk 1]
```diff
@@ -78,6 +79,17 @@ contract MToken is IMToken, ContinuousIndexing, ERC20Extended {
 
     /// @inheritdoc IMToken
     function burn(address account_, uint256 amount_) external onlyPortal {
+        uint256 spenderAllowance_ = allowance[account_][msg.sender];
```
Comment (toninorair) [inline src/MToken.sol:82]: what happens if an account is a `msg.sender` itself? It won't have an allowance set, and won't be able to proceed with the call.

Comment (toninorair) [reply src/MToken.sol:82]: resolved in conversation. This allowance feature is only for Portal

Comment (toninorair) [inline src/MToken.sol:82]:
```suggestion
        uint256 portalAllowance_ = allowance[account_][portal];
```

Comment (PierrickGT) [reply src/MToken.sol]: Fixed in this commit: https://github.com/m0-foundation/protocol/pull/190/commits/ed31631c7f2fc0ed50176cd0f46df6c91d9aa9d2

**Review by toninorair — APPROVED** [review]
lgtm

**Pull request merged**